lib/structures/Interaction.js

```diff
@@ -0,0 +1,248 @@
+"use strict";
+
+const Base = require("./Base");
+const Message = require("./Message");
+const Member = require("./Member");
+const Constants = require("../Constants");
+
+/**
+* Represents an interaction
+* @prop {String} applicationId The id of the interaction's application
+* @prop {String?} channelId The interaction token
+* @prop {Object?} data The data attached to the interaction
+* @prop {Number?} data.componentType The type of Message Component (Message Component only)
+* @prop {String?} data.id The ID of the Slash Command (Slash Command)
+* @prop {String?} data.custom_id The ID of the Button (Message Component)
+* @prop {String?} data.name The command name (Slash Command only)
+* @prop {Array<Object>?} data.options The run Slash Command options (Slash Command only)
+* @prop {String?} data.options.name The name of the Slash Command option (Slash Command only)
+* @prop {String?} data.options.value The value of the run Slash Command (Slash Command only)
+* @prop {String?} guildId The id of the guild in which the interaction was created
+* @prop {String} id The interaction id
+* @prop {Member?} member the member who triggered the interaction
+* @prop {Message?} message The message the interaction came from (Message Component only)
+* @prop {String} token The interaction token (Interaction tokens are valid for 15 minutes and can be used to send followup messages but you must send an initial response within 3 seconds of receiving the event. If the 3 second deadline is exceeded, the token will be invalidated.)
+* @prop {Number} type 1 is a Ping, 2 is a Slash Command, 3 is a Message Component
+* @prop {Number} version The interaction version
+*/
+class Interaction extends Base {
+    constructor(data, client) {
+        super(data.id);
+        this._client = client;
+
+        this.applicationId = data.application_id;
+
+        if(data.channel_id !== undefined) {
+            this.channelId = data.channel_id;
+        }
+
+        if(data.data !== undefined) {
+            this.data = data.data;
+        }
+
+        if(data.guild_id !== undefined) {
+            this.guildId = data.guild_id;
+        }
+
+        if(data.member !== undefined) {
+            this.member = new Member(data.member, client.guilds.get(data.guild_id), this._client);
+        }
+
+        if(data.message !== undefined) {
+            this.message = new Message(data.message, this._client);
+        }
+
+        if(data.token !== undefined) {
+            this.token = data.token;
+        }
+
+        this.type = data.type;
+        this.version = data.version;
+    }
+
+    /**
+    * Note: You can **not** use more then 1 initial interaction response per interaction.
```

Comment:
shouldn't it be "than" instead of "then"

Reply:
Yea you are right